status-im/status-desktop
# Feat/request payment modal 16736

Closes #16736

### What does the PR do

* Added feature flag for Request Payment. Disabled until whole feature is finished
* Added Request payment modal
* Added menu for message commands (now featuring add image and add request payment). Request payment is disabled for testnet to prevent corner cases or issues.
* Added RequestPaymentStore to easily pass required models and stores to modal
* Updated storybook

### Affected areas



### Architecture compliance

My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)


![image](https://github.com/user-attachments/assets/8ec884b6-924c-481b-b883-71ee8d2b9b07)

<img width="1130" alt="image" src="https://github.com/user-attachments/assets/00512d3e-1675-464f-81cf-16e0b51c2634">

### Impact on end user

What is the impact of these changes on the end user (before/after behaviour)

### How to test

- How should one proceed with testing this PR.
Enable request payment feature flag, go to any chat on mainnet.
- What kind of user flows should be checked?

### Risk 

Described potential risks and worst case scenarios.

Tick **one**:
